Dockerfile

```diff
@@ -204,7 +204,12 @@ RUN pip install -e .
 RUN tethys gen portal_config
 
 # Install channel-redis
-RUN pip install channels_redis
+RUN if [[ ${DJANGO_VERSION} =~ ^3 ]]; then \
+  touch ~/testfile.txt; \
+  pip install channels==4.0.0 channels_redis; \
+  else \
+  pip install channels_redis; \
```

Comment:
Starting to think the real issue is that pip doesn't honor the pins set by conda when the Conda environment is created. Can you try installing channels_redis with conda instead of pip to see if it respects the Django pin and resolves the correct channels?

Reply:
Updated to use micromamba installer.